Dockerfile.server

```diff
@@ -0,0 +1,14 @@
+FROM golang:1.21 AS builder
```

Comment:
lets add this to gha action to push this image too
simmilar to the binary image
https://hub.docker.com/repository/docker/medyagh/gopogh/general

https://github.com/medyagh/gopogh/blob/master/.github/workflows/image_build.yml